docs/source/scale/sagemaker.rst

```diff
@@ -1,17 +1,398 @@
 .. _distributed-sagemaker:
 
-[**Under construction**]
+Use GraphStorm on SageMaker
+===================================
+GraphStorm can run on Amazon Sagemaker to leverage SageMaker's ML DevOps capabilities.
 
-Use GraphStorm in SageMaker
-============================
-
-Setup SageMaker
+Prerequisites
 -----------------
+In order to use GraphStorm on Amazon SageMaker, users need to have AWS access to the following AWS services.
 
+- **SageMaker service**. Please refer to `Amazon SageMaker service <https://aws.amazon.com/pm/sagemaker/>`_ for how to get access to Amazon SageMaker.
+- **Amazon ECR**. Please refer to `Amazon Elastic Container Registry service <https://aws.amazon.com/ecr/>`_ for how to get access to Amazon ECR.
+- **S3 service**. Please refer to `Amazon S3 service <https://aws.amazon.com/s3/>`_ for how to get access to Amazon S3.
+- **SageMaker Framework Containers**. Please follow `AWS Deep Learning Containers guideline <https://github.com/aws/deep-learning-containers>`_ to get access to the image.
+- **Amazon EC2** (optional). Please refer to `Amazon EC2 service <https://aws.amazon.com/ec2/>`_ for how to get access to Amazon EC2.
 
-Launch Training
------------------
+Setup GraphStorm SageMaker Docker Image
+----------------------------------------------
+GraphStorm uses SageMaker's **BYOC** (Bring Your Own Container) mode. Therefore, before launching GraphStorm on SageMaker, there are two steps required to setup a GraphStorm SageMaker Docker image.
+
+.. _build_sagemaker_docker:
+
+Step 1: Build a SageMaker-compatible Docker image
+...................................................
+
+.. note::
+    * Please make sure your account has access key (AK) and security access key (SK) configured to authenticate accesses to AWS services.
+    * For more details of Amazon ECR operation via CLI, users can refer to the `Using Amazon ECR with the AWS CLI document <https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-cli.html>`_.
+
+First, in a Linux machine, configure a Docker environment by following the `Docker documentation <https://docs.docker.com/get-docker/>`_ suggestions.
+
+In order to use the SageMaker base Docker image, users need to use the following command to authenticate to pull SageMaker images.
+
+.. code-block:: bash
+
+    aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 763104351884.dkr.ecr.us-east-1.amazonaws.com
```

Comment:
```suggestion
    aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 888888888888.dkr.ecr.us-east-1.amazonaws.com
```

Reply:
This account id is the SageMaker base image and is public. So cannot change to 88888888888. Otherwise this command will fail.